rclrs/src/context.rs

```diff
@@ -138,7 +138,30 @@ impl Context {
         node_namespace: &str,
         node_name: &str,
     ) -> Result<Node, RclReturnCode> {
-        Node::new_with_namespace(node_namespace, node_name, self)
+        NodeBuilder::new(node_name, self)
+            .namespace(node_namespace)
+            .build()
+    }
+
+    /// Creates a [`NodeBuilder`][1].
+    ///
+    /// Convenience function equivalent to [`NodeBuilder::new`][2].
+    /// Please see that function's documentation.
+    ///
+    /// [1]: crate::NodeBuilder
+    /// [2]: crate::NodeBuilder::new
+    ///
+    /// # Example
+    /// ```
+    /// # use rclrs::{Context, RclReturnCode};
+    /// let context = Context::new([])?;
+    /// let node_builder = context.create_node_builder("foo_node");
+    /// let node = node_builder.build()?;
+    /// assert_eq!(node.name(), "foo_node");
+    /// # Ok::<(), RclReturnCode>(())
+    /// ```
+    pub fn create_node_builder(&self, node_name: &str) -> NodeBuilder {
```

Comment:
In the same way as before, `NodeBuilder`s should be created via the `Context`. Moreover, `rclcpp` uses `NodeFactory` to instantiate `Node`s, so we could replicate that API to make it more conventional.